tcms/testplans/static/testplans/js/get.js

```diff
@@ -19,26 +19,39 @@ $(document).ready(function() {
                 testCases.not_confirmed.push(data[i]);
             }
         }
-        drawConfirmedTestCases();
+        drawConfirmedTestCases(testCases.confirmed);
     });
 
-    function drawConfirmedTestCases() {
-        var container = $('#confirmed-testcases'),
-            noCasesTemplate = $('#no_test_cases'),
-            testCaseRow = $('#test_case_row')[0],
-            testCaseSummary = $(testCaseRow.content).find('.list-view-pf-main-info a.testcase')[0];
-
-        if (testCases.confirmed.length > 0) {
-            for (var i = 0; i < testCases.confirmed.length; i++) {
-                var testCase = testCases.confirmed[i];
-                testCaseSummary.href = `/case/${testCase.id}/`;
-                testCaseSummary.innerHTML = `TC-${testCase.id}: ${testCase.summary}`;
-                container.append($(testCaseRow).html());
-            }
-        } else {
-            container.append(noCasesTemplate[0].innerHTML);
-        }
+});
 
-        treeViewBind();
+function drawConfirmedTestCases(testCases) {
+    var container = $('#confirmed-testcases'),
+        noCasesTemplate = $('#no_test_cases'),
+        testCaseRow = $('#test_case_row')[0],
+        testCaseSummary = $(testCaseRow.content).find('.js-test-case-summary')[0];
+
+    if (testCases.length > 0) {
+        for (var i = 0; i < testCases.length; i++) {
+            var testCase = testCases[i];
+            testCaseSummary.innerHTML = getTestCaseRowContent(testCase);
+            container.append($(testCaseRow).html());
+        }
+    } else {
+        container.append(noCasesTemplate[0].innerHTML);
     }
-});
+
+    treeViewBind();
+}
+
+
+function getTestCaseRowContent(testCase) {
+    var rowDetailsTemplate = $('#test_case_row_details')[0],
+        row = $(rowDetailsTemplate.content);
```

Comment:
Move the details HTML inside the row HTML. This will make understanding and changing the HTML structure more easier. You already have classes to find & replace the actual values so you just need to pass `row` as parameter here.